### PR TITLE
kubelet: detect process overlap during apiserver rollout

### PR DESCRIPTION
**Pull request by mfojtik**
https://github.com/openshift/kubernetes/pull/876/files introduced a detection mechanism that uses an event when there are two overlapping processes running of apiserver. This should never happen and if it happens it should become a blocker bug for the release to fix it.

### CODE REVIEW

File: pkg/synthetictests/event_junits.go [hunk 1]
```diff
@@ -41,6 +42,7 @@ func SystemUpgradeEventInvariants(events monitorapi.Intervals, duration time.Dur
 	tests = SystemEventInvariants(events, duration, kubeClientConfig)
 	tests = append(tests, testContainerFailures(events)...)
 	tests = append(tests, testDeleteGracePeriodZero(events)...)
+	tests = append(tests, testKubeApiserverProcessOverlap(events)...)
```
Comment (deads2k) [inline pkg/synthetictests/event_junits.go:45]: this looks like a systemeventinvariant to me.  no matter what, we shouldn't have two processes running.

Comment (mfojtik) [reply pkg/synthetictests/event_junits.go:45]: @deads2k it exists in both

File: pkg/synthetictests/kubelet.go [hunk 2]
```diff
@@ -135,6 +135,26 @@ func testContainerFailures(events monitorapi.Intervals) []*ginkgo.JUnitTestCase
...
+		if strings.Contains(event.Message, "reason/TerminationProcessOverlapDetected") {
+			failures = append(failures, event.Locator)
+		}
+	}
+
+	failure := &ginkgo.JUnitTestCase{
+		Name:      testName,
+		SystemOut: strings.Join(failures, "\n"),
+		FailureOutput: &ginkgo.FailureOutput{
+			Output: fmt.Sprintf("The following events detected:\n\n%s", strings.Join(failures, "\n")),
```
Comment (deads2k) [inline pkg/synthetictests/kubelet.go:152]: you're going to want to display the time as well.

Comment (mfojtik) [reply pkg/synthetictests/kubelet.go:152]: fixed

**Comment by mfojtik** [comment]
adding valid-bug manually as this is a test and does not require QE.

**Pull request merged**